monkey/infection_monkey/model/host.py

```diff
@@ -1,17 +1,24 @@
 from ipaddress import IPv4Address
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, Union
 
 from pydantic import Field
+from typing_extensions import Literal  # import from `typing` once we switch to Python 3.8
 
 from common import OperatingSystem
 from common.base_models import MutableInfectionMonkeyBaseModel
+from common.types import NetworkPort, NetworkProtocol
+from infection_monkey.dataclasses import PortScanData
 
 
 class TargetHost(MutableInfectionMonkeyBaseModel):
     ip: IPv4Address
     operating_system: Optional[OperatingSystem] = Field(default=None)
     services: Dict[str, Any] = Field(default={})  # deprecated
     icmp: bool = Field(default=False)
+    port_status: Dict[
+        Union[Literal[NetworkProtocol.TCP], Literal[NetworkProtocol.UDP]],
+        Dict[NetworkPort, PortScanData],
+    ] = Field(default={})
```

Comment:
Hmm. I'm rethinking this design choice. This is on me.

Instead of a nested dict, it would probably be cleaner if we had pydantic model called `PortStatus` with two fields, `tcp_ports` and `udp_ports`. This would give us nearly the same level of extensibility but result in cleaner code, since we don't need to deal with nested dicts.

We might then consider adding `NetworkProtocol` to `PortScanData` for completeness, even if we only ever set it to `TCP` for now.

Reply:
We'll need a different name since we already have a `PortStatus` enum, but makes sense.

Reply:
I propose `TargetHostPorts`. I'm open to other suggestions.